src/gamecontroller.cs

```diff
@@ -535,4 +621,1162 @@ interface GCEventViewController {
+		NSSet<GCControllerAxisInput> AllAxes { get; }
+
+		[Export ("allDpads", ArgumentSemantic.Strong)]
+		NSSet<GCControllerDirectionPad> AllDpads { get; }
+
+		[Export ("objectForKeyedSubscript:")]
+		[return: NullAllowed]
+		GCControllerElement ObjectForKeyedSubscript (string key);
+
+		[Export ("capture")]
+		GCPhysicalInputProfile Capture ();
+
+		[Export ("setStateFromPhysicalInput:")]
+		void SetState (GCPhysicalInputProfile physicalInput);
+	}
+
+	// These constants could go inside of GCPhysicalInputProfile or GCController, thoughts?
+	[TV (14, 0), Mac (11, 0), iOS (14, 0)]
+	[Static]
+	interface GCInputXbox {
+
+		[Field ("GCInputXboxPaddleOne")]
+		NSString PaddleOne { get; }
+
+		[Field ("GCInputXboxPaddleTwo")]
+		NSString PaddleTwo { get; }
+
+		[Field ("GCInputXboxPaddleThree")]
+		NSString PaddleThree { get; }
+
+		[Field ("GCInputXboxPaddleFour")]
+		NSString PaddleFour { get; }
+	}
+
+	[TV (14, 0), Mac (11, 0), iOS (14, 0)]
+	[Static]
+	interface GCInput {
+
+		[Field ("GCInputButtonA")]
+		NSString ButtonA { get; }
+
+		[Field ("GCInputButtonB")]
+		NSString ButtonB { get; }
+
+		[Field ("GCInputButtonX")]
+		NSString ButtonX { get; }
+
+		[Field ("GCInputButtonY")]
+		NSString ButtonY { get; }
+
+		[Field ("GCInputDirectionPad")]
+		NSString DirectionPad { get; }
+
+		[Field ("GCInputLeftThumbstick")]
+		NSString LeftThumbstick { get; }
+
+		[Field ("GCInputRightThumbstick")]
+		NSString RightThumbstick { get; }
+
+		[Field ("GCInputLeftShoulder")]
+		NSString LeftShoulder { get; }
+
+		[Field ("GCInputRightShoulder")]
+		NSString RightShoulder { get; }
+
+		[Field ("GCInputLeftTrigger")]
+		NSString LeftTrigger { get; }
+
+		[Field ("GCInputRightTrigger")]
+		NSString RightTrigger { get; }
+
+		[Field ("GCInputLeftThumbstickButton")]
+		NSString LeftThumbstickButton { get; }
+
+		[Field ("GCInputRightThumbstickButton")]
+		NSString RightThumbstickButton { get; }
+
+		[Field ("GCInputButtonHome")]
+		NSString ButtonHome { get; }
+
+		[Field ("GCInputButtonMenu")]
+		NSString ButtonMenu { get; }
+
+		[Field ("GCInputButtonOptions")]
+		NSString ButtonOptions { get; }
+
+		[Field ("GCInputDualShockTouchpadOne")]
+		NSString DualShockTouchpadOne { get; }
+
+		[Field ("GCInputDualShockTouchpadTwo")]
+		NSString DualShockTouchpadTwo { get; }
+
+		[Field ("GCInputDualShockTouchpadButton")]
+		NSString DualShockTouchpadButton { get; }
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (GCExtendedGamepad))]
+	interface GCXboxGamepad
+	{
+		[NullAllowed, Export ("paddleButton1")]
+		GCControllerButtonInput PaddleButton1 { get; }
+
+		[NullAllowed, Export ("paddleButton2")]
+		GCControllerButtonInput PaddleButton2 { get; }
+
+		[NullAllowed, Export ("paddleButton3")]
+		GCControllerButtonInput PaddleButton3 { get; }
+
+		[NullAllowed, Export ("paddleButton4")]
+		GCControllerButtonInput PaddleButton4 { get; }
+	}
+
+	[Static] // I tried moving GCKey and GCKeyCode to separate files, but ran into
```

Comment:
Is there a way for `GCKey` and `GCKeyCode` to live in their own files?

Reply:
It's possible, but it's somewhat of a special case.

Basically you'd have to add it like xkit.cs is added here:

https://github.com/xamarin/xamarin-macios/blob/53dcedc0609b6f0fa18cdbb1484bb948b94159d3/src/frameworks.sources#L1799

I don't think it's worth it to have binding code in a separate file, xkit.cs is literally the only exception, and that's only because we want the code in both UIKit and AppKit at the same time.